sharding-core/sharding-core-api/src/main/java/org/apache/shardingsphere/api/config/sharding/KeyGeneratorConfiguration.java

```diff
@@ -20,27 +20,21 @@
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
 import lombok.Getter;
-import org.apache.shardingsphere.underlying.common.config.TypeBasedSPIConfiguration;
-
-import java.util.Properties;
+import org.apache.shardingsphere.spi.keygen.ShardingKeyGenerator;
 
 /**
  * Key generator configuration.
 */
 @Getter
-public final class KeyGeneratorConfiguration extends TypeBasedSPIConfiguration {
+public final class KeyGeneratorConfiguration {
     
     private final String column;
     
-    public KeyGeneratorConfiguration(final String type, final String column) {
-        super(type);
-        Preconditions.checkArgument(!Strings.isNullOrEmpty(column), "Column is required.");
-        this.column = column;
-    }
+    private final ShardingKeyGenerator keyGenerator;
```

Comment:
here we should rename ShardingKeyGenerator SPI to KeyGenerateAlgorithm.
Be care about modifying all sub class of ShardingKeyGenerator

Reply:
ShardingKeyGenerator has renamed to KeyGenerateAlgorithm.

Reply:
> here we should rename ShardingKeyGenerator SPI to KeyGenerateAlgorithm.
> Be care about modifying all sub class of ShardingKeyGenerator

ShardingKeyGenerator has renamed to KeyGenerateAlgorithm.

Reply:
> here we should rename ShardingKeyGenerator SPI to KeyGenerateAlgorithm.
> Be care about modifying all sub class of ShardingKeyGenerator

ShardingKeyGenerator has renamed to KeyGenerateAlgorithm, including all sub class of ShardingKeyGenerator and ShardingKeyGeneratorServiceLoader.